src/main/java/com/questhelper/helpers/quests/perilousmoon/PerilousMoon.java

```diff
@@ -0,0 +1,702 @@
+		steps.put(23, goGetItems);
+		steps.put(24, goGetItems);
+		steps.put(25, goGetItems);
+		steps.put(26, goGetItems);
+
+		ConditionalStep goTalkToEyatlalliAfterItems = goEnterNeypotzli.copy();
+		goTalkToEyatlalliAfterItems.addStep(inNeypotzli, talkToEyatlalliAfterItems);
+		steps.put(27, goTalkToEyatlalliAfterItems);
+
+		ConditionalStep goFightBlueMoon = new ConditionalStep(this, enterEarthboundForFight);
+		goFightBlueMoon.addStep(inBlueMoon, fightBlueMoon);
+		goFightBlueMoon.addStep(inEarthbound, enterBlueMoonFight);
+
+		ConditionalStep goFightBloodMoon = new ConditionalStep(this, enterPrisonForFight);
+		goFightBloodMoon.addStep(inBloodMoon, fightBloodMoon);
+		goFightBloodMoon.addStep(inPrison, enterBloodMoonFight);
+
+		ConditionalStep goFightEclipseMoon = new ConditionalStep(this, enterStreamboundForFight);
+		goFightEclipseMoon.addStep(inEclipseMoon, fightEclipseMoon);
+		goFightEclipseMoon.addStep(inStreambound, enterEclipseMoonFight);
+
+		ConditionalStep goDefeatMoons = new ConditionalStep(this, enterCamTorumForFights);
+		goDefeatMoons.addStep(and(inNeypotzli, defeatedBlueMoon, defeatedBloodMoon, defeatedEclipseMoon), talkToJessamineEnd);
+		goDefeatMoons.addStep(and(inNeypotzli, defeatedBlueMoon, defeatedBloodMoon), goFightEclipseMoon);
+		goDefeatMoons.addStep(and(inNeypotzli, defeatedBlueMoon), goFightBloodMoon);
+		goDefeatMoons.addStep(inNeypotzli, goFightBlueMoon);
+		goDefeatMoons.addStep(inCamTorum, enterNeypotzliForFights);
+
+		steps.put(28, goDefeatMoons);
+		steps.put(29, goDefeatMoons);
+
+		ConditionalStep goFinishWithZuma = goEnterNeypotzli.copy();
+		goFinishWithZuma.addStep(inNeypotzli, talkToZumaEnd);
+		steps.put(30, goFinishWithZuma);
+
+		ConditionalStep goFinishQuest = goEnterNeypotzli.copy();
+		goFinishQuest.addStep(inNeypotzli, finishQuest);
+		steps.put(31, goFinishQuest);
+
+		return steps;
+	}
+
+	private void setupZones()
+	{
+		camTorum = new Zone(new WorldPoint(1378, 9502, 1), new WorldPoint(1524, 9600, 3));
+		antechamber = new Zone(5782, 1);
+		prison = new Zone(5525, 0);
+		earthbound = new Zone(5527, 0);
+		streambound = new Zone(6039, 0);
+		ancientShrine = new Zone(6037, 0);
+
+		blueMoonRoom = new Zone(new WorldPoint(1417, 9647, 0), new WorldPoint(1463, 9699, 0));
+		bloodMoonRoom = new Zone(new WorldPoint(1373, 9614, 0), new WorldPoint(1421, 9650, 0));
+		eclipseRoom = new Zone(new WorldPoint(1457, 9614, 0), new WorldPoint(1506, 9650, 0));
+	}
+
+	@Override
+	public void setupRequirements()
+	{
+		knife = new ItemRequirement("Knife", ItemID.KNIFE).isNotConsumed();
+		knife.canBeObtainedDuringQuest();
+		bigFishingNet = new ItemRequirement("Big fishing net", ItemID.BIG_FISHING_NET).isNotConsumed();
+		bigFishingNet.canBeObtainedDuringQuest();
+		rope = new ItemRequirement("Rope", ItemID.ROPE);
+		rope.canBeObtainedDuringQuest();
+		pestleAndMortar = new ItemRequirement("Pestle and mortar", ItemID.PESTLE_AND_MORTAR).isNotConsumed();
+		pestleAndMortar.canBeObtainedDuringQuest();
```

Comment:
This lets us give the player a hint if they've been silly & dropped things
```suggestion
		knife = new ItemRequirement("Knife", ItemID.KNIFE).isNotConsumed();
		knife.setTooltip("Can be obtained during the quest from the Supply crates (fishing supplies).");
		bigFishingNet = new ItemRequirement("Big fishing net", ItemID.BIG_FISHING_NET).isNotConsumed();
		bigFishingNet.setTooltip("Can be obtained during the quest from the Supply crates (fishing supplies).");
		rope = new ItemRequirement("Rope", ItemID.ROPE);
		rope.setTooltip("Can be obtained during the quest from the Supply crates (hunting supplies).");
		pestleAndMortar = new ItemRequirement("Pestle and mortar", ItemID.PESTLE_AND_MORTAR).isNotConsumed();
		pestleAndMortar.setTooltip("Can be obtained during the quest from the Supply crates (herblore supplies).");
```